utils/lib/nr-graphql-helpers.ts

```diff
@@ -175,6 +175,32 @@ export const getCategoryTermsFromKeywords = async (
   return categoryKeywords.length > 0 ? categoryKeywords : undefined;
 };
 
+type CoreDataSourceSearchResults = {
+  actor: {
+    nr1Catalog: {
+      search: {
+        results: {
+          id: string;
+        }[];
+      };
+    };
+  };
+};
+
+export const getCoreDataSourceIds = async (): Promise<string[]> => {
+  const { data } = await fetchNRGraphqlResults<{}, CoreDataSourceSearchResults>(
+    { queryString: CORE_DATA_SOURCES_QUERY, variables: {} }
+  );
```

Comment:
This function can possibly return errors, so we should handle those errors if we get them. 
I think you have a few options here, we could either throw the error and reject the promise, or just log out any errors and return an empty array.